### PR TITLE
[Sprint 4] Feature add book logic 2

### PR DESCRIPTION
**Pull request by nguyenminhduong**
Redmine https://dev.framgia.com/redmine/issues/136395
Feature add book logic

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -61,4 +61,5 @@
     <string name="name">Name</string>
     <string name="following">Following</string>
     <string name="select_image_from_gallery_or_camera">Select image from gallery or camera</string>
+    <string name="select_office">Select office</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`select_office`" is not translated in "vi" (Vietnamese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: WARNING: [UnusedResources] The resource `R.string.select_office` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/java/com/framgia/fbook/screen/sharebook/ShareBookActivity.kt [hunk 2]
```diff
@@ -88,6 +98,27 @@ class ShareBookActivity : BaseActivity(), ShareBookContract.ViewModel, ItemImage
...
+
+  override fun onGetOfficeSuccess(listOffice: List<Office>?) {
+    mListOffice.addAll(listOffice!!)
+    setCurrentOffice()
+  }
+
+  override fun onError(baseException: BaseException) {
+  }
+
+  private fun setCurrentOffice() {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/fbook/screen/sharebook/ShareBookActivity.kt]: pls change to ``updateCurrentOffice(int currentOfficePosition)``
Avoid using global variable inside method private

File: app/src/main/java/com/framgia/fbook/screen/sharebook/ShareBookActivity.kt [hunk 3]
```diff
@@ -25,12 +29,17 @@ class ShareBookActivity : BaseActivity(), ShareBookContract.ViewModel, ItemImage
...
+  @Inject
   internal lateinit var mImageSelectedAdapter: ImageSelectedAdapter
   val mBookRequest = BookRequest()
+  private val mListOffice = mutableListOf<Office>()
 
   val titleError: ObservableField<String> = ObservableField()
   val authorError: ObservableField<String> = ObservableField()
   val descriptionError: ObservableField<String> = ObservableField()
+  var currentOffice: ObservableField<String> = ObservableField()
+  var currentOfficePosition: Int = 0
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/fbook/screen/sharebook/ShareBookActivity.kt]: can be private

**Pull request merged**